mesonbuild/environment.py

```diff
@@ -931,6 +931,10 @@ def detect_java_compiler(self):
             raise EnvironmentException('Could not execute Java compiler "%s"' % ' '.join(exelist))
         if 'javac' in out or 'javac' in err:
             version = search_version(err if 'javac' in err else out)
+            if not version or version == 'unknown version':
+                parts = (err if 'javac' in err else out).split()
+                if len(parts) > 1:
+                    version = parts[1]
```

Comment:
I wonder if we'd be better of not using `search_version` here at all. Or maybe we should just fix it to not be unmaintainable regex.

Reply:
Skipping the call to search_function was my original idea in Issue #4322, but I changed to only working around when necessary on the principle of minimal changes to existing code. As regards "unmaintainable regex", please see my Issue #5020 which uses a verbose RE compile to make the meaning more evident.  As for changing the regex so it doesn't require a period in the version string, I experimented with that but realized that it was just too eager to match anything with a digit in it. For example, it could match '69' in "javac -version" output "java69 13'.

Reply:
Sorry, meant call to search_version, not search_function.